core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -428,15 +479,57 @@ private[cassandra] object CassandraJournal {
   private case class MessageId(persistenceId: String, sequenceNr: Long)
   private case class SerializedAtomicWrite(persistenceId: String, payload: Seq[Serialized])
   private case class Serialized(persistenceId: String, sequenceNr: Long, serialized: ByteBuffer, tags: Set[String],
-                                eventManifest: String, serManifest: String, serId: Int, writerUuid: String)
+                                eventAdapterManifest: String, serManifest: String, serId: Int, writerUuid: String,
+                                meta: Option[SerializedMeta])
+  private case class SerializedMeta(serialized: ByteBuffer, serManifest: String, serId: Int)
   private case class PartitionInfo(partitionNr: Long, minSequenceNr: Long, maxSequenceNr: Long)
 
-  def deserializeEvent(serialization: Serialization, row: Row): Any = {
-    serialization.deserialize(
-      row.getBytes("event").array,
-      row.getInt("ser_id"),
-      row.getString("ser_manifest")
-    ).get
+  class EventDeserializer(serialization: Serialization) {
+
+    // cache to avoid repeated check via ColumnDefinitions
+    @volatile private var _hasMetaColumns: Option[Boolean] = None
+
+    def hasMetaColumns(row: Row): Boolean = _hasMetaColumns match {
+      case Some(b) => b
+      case None =>
+        val b = row.getColumnDefinitions.contains("meta")
+        _hasMetaColumns = Some(b)
+        b
+    }
+
+    def deserializeEvent(row: Row): Any = {
+      val event = serialization.deserialize(
+        row.getBytes("event").array,
```

Comment:
currently it works, but it's a valid point and I have created a separate issue https://github.com/akka/akka-persistence-cassandra/issues/225